pkg/util/scheduler/queue_test.go

```diff
@@ -0,0 +1,635 @@
+		cancel()
+		require.ErrorIs(t, err, context.DeadlineExceeded, "Dequeue on empty queue should time out")
+		require.False(t, ok, "Dequeue on empty queue should return ok=false")
+
+		// Stop the queue
+		qSimple.StopAsync()
+		require.NoError(t, qSimple.AwaitTerminated(context.Background()), "Queue should stop")
+	})
+
+	t.Run("RoundRobinBetweenTenants", func(t *testing.T) {
+		t.Parallel()
+
+		ctx := context.Background()
+		q := NewQueue(QueueOptionsWithDefaults(nil))
+		require.NoError(t, q.StartAsync(ctx), "Queue should start")
+		require.NoError(t, q.AwaitRunning(ctx), "Queue should be running")
+
+		tenantA := "tenant-a"
+		tenantB := "tenant-b"
+
+		// We'll use a very small test with just 2 items per tenant
+		// to reduce the chance of timeouts or other issues
+
+		// Enqueue items in tenant order: A, B, A, B
+		// Each item will record its tenant ID when executed
+		var results []string
+		var resultsMu sync.Mutex
+
+		makeRunnable := func(id string) func() {
+			return func() {
+				resultsMu.Lock()
+				results = append(results, id)
+				resultsMu.Unlock()
+			}
+		}
+
+		// First tenant A, then B, then A, then B
+		err := q.Enqueue(ctx, tenantA, makeRunnable(tenantA))
+		require.NoError(t, err)
+		err = q.Enqueue(ctx, tenantB, makeRunnable(tenantB))
+		require.NoError(t, err)
+		err = q.Enqueue(ctx, tenantA, makeRunnable(tenantA))
+		require.NoError(t, err)
+		err = q.Enqueue(ctx, tenantB, makeRunnable(tenantB))
+		require.NoError(t, err)
+
+		// Verify queue state
+		require.Equal(t, 4, q.Len(), "Queue should have 4 items")
+		require.Equal(t, 2, q.ActiveTenantsLen(), "Should have 2 active tenants")
+
+		// Use a longer timeout to handle CI environment variability
+		dequeueTimeout := 3 * time.Second
+
+		// Dequeue and execute the four items
+		for i := 0; i < 4; i++ {
+			// Use a more reliable context with longer timeout for CI environments
+			dequeueCtx, cancel := context.WithTimeout(ctx, dequeueTimeout)
+			runnable, ok, err := q.Dequeue(dequeueCtx)
+			if !ok || err != nil {
+				// If there's an error, let's check if queue state is still consistent
+				t.Logf("Dequeue %d failed: ok=%v, err=%v", i, ok, err)
+				t.Logf("Queue state: Len=%d, ActiveTenantsLen=%d", q.Len(), q.ActiveTenantsLen())
+			}
+
+			cancel()
+			require.NoError(t, err, "Dequeue %d should succeed", i)
+			require.True(t, ok, "Dequeue %d should return ok=true", i)
+			require.NotNil(t, runnable, "Dequeued runnable %d should not be nil", i)
+			runnable() // Execute to record the tenant ID
+		}
+
+		// Check execution order - should alternate between tenants
+		resultsMu.Lock()
+		expectedOrder := []string{tenantA, tenantB, tenantA, tenantB}
+		equal := len(results) == len(expectedOrder)
+		if equal {
+			for i, v := range expectedOrder {
+				if i >= len(results) || results[i] != v {
+					equal = false
+					break
+				}
+			}
+		}
+
+		if !equal {
+			t.Errorf("Execution order mismatch: expected %v, got %v", expectedOrder, results)
+		}
```

Comment:
Previous ~10 lines can be replaced with 

```suggestion
		require.Equal(t, []string{tenantA, tenantB, tenantA, tenantB}, results)
```